src/v3/src/transformer/selectAuthenticator/transformSelectAuthenticatorEnroll.test.ts

```diff
@@ -190,4 +212,288 @@ describe('Enroll Authenticator Selector Transformer Tests', () => {
     expect((updatedFormBag.uischema.elements[4] as ButtonElement).options.type)
       .toBe(ButtonType.SUBMIT);
   });
+
+  it('should transform authenticator elements when all elements have non-expired grace periods', () => {
+    const options = [
+      {
+        label: 'Email',
+        value: 'okta_email',
+        relatesTo: {
+          gracePeriod: {
+            expiry: '2045-09-27 18:00:00.000.',
+          },
+        },
+      } as unknown as IdxOption,
+      {
+        label: 'Email',
+        value: 'okta_email',
+        relatesTo: {
+          gracePeriod: {
+            expiry: '2030-09-27 18:00:00.000.',
```

Comment:
I assume you are using far in the future expiration dates to try to guarantee that this test will work as you expect. I would suggest we try to use mocking of `new Date` and `Date.now` whenever possible to create realistic unit test scenarios that are also stable over the long run.

P.S. Are you not expecting this test to need to pass after Sept 2030? 😛

Reply:
The same comment also applies to other mocked date objects in this PR, including ones that are in the past. We should still mock the value of the present date so that snapshots and assertions are unchanging.

Reply:
😄 I think this file is the only one we missed that depends on the present date? Let me know if we missed any other test.

We had already mocked `Date.now` in testcafe here: https://github.com/okta/okta-signin-widget/pull/3769/files#diff-2900b8cd411a393bd492a86b3fa5c19025d50ea6928545d641ea0341fbfacaf2R86

and v2 unit tests here: https://github.com/okta/okta-signin-widget/pull/3769/files#diff-2af053911968b4d4959d3566bd261769a8cc5324853397eb90b51f5b9052aed8R29